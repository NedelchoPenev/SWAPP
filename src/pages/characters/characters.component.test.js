import React from 'react';
import { shallow } from 'enzyme';

import CharactersPage from './characters.component';
import CustomButton from '../../components/custom-button/custom-button.component';

describe('CharactersPage component', () => {

  it('should render CharactersPage component', () => {
    const mockProps = {
      people: {
        edges: [{
          node: {
            id: 1
          }
        }],
        pageInfo: {
          hasNextPage: true
        }
      }
    }
    expect(shallow(<CharactersPage {...mockProps}/>)).toMatchSnapshot();
  });

  it('should NOT render CustomButton component', () => {
    const mockProps = {
      people: {
        edges: [{
          node: {
            id: 1
          }
        }],
        pageInfo: {
          hasNextPage: false
        }
      }
    }
    expect(shallow(<CharactersPage {...mockProps}/>).find(CustomButton)).toEqual({});
  });
});