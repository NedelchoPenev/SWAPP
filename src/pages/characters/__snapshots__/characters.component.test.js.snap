// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersPage component should render CharactersPage component 1`] = `
<styled.div>
  <styled.div>
    <withRouter(CharacterCard)
      character={
        Object {
          "id": 1,
        }
      }
      key="1"
    />
  </styled.div>
  <CustomButton>
     Load More 
  </CustomButton>
</styled.div>
`;
